src/wayland/buffer.rs

```diff
@@ -15,6 +14,49 @@ use wayland_client::{
 
 use super::State;
 
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
+fn create_memfile() -> File {
+    use nix::sys::memfd::{memfd_create, MemFdCreateFlag};
+
+    let name = unsafe { CStr::from_bytes_with_nul_unchecked("softbuffer\0".as_bytes()) };
+    let fd = memfd_create(name, MemFdCreateFlag::MFD_CLOEXEC)
+        .expect("Failed to create memfd to store buffer.");
+    unsafe { File::from_raw_fd(fd) }
+}
+
+#[cfg(not(any(target_os = "linux", target_os = "freebsd")))]
+fn create_memfile() -> File {
+    use nix::{
+        errno::Errno,
+        fcntl::OFlag,
+        sys::{
+            mman::{shm_open, shm_unlink},
+            stat::Mode,
+        },
+    };
+    use rand::distributions::{Alphanumeric, DistString};
+
+    for _ in 0..=4 {
+        let mut name = String::from("softbuffer-");
+        Alphanumeric.append_string(&mut rand::thread_rng(), &mut name, 7);
```

Comment:
This feels overkill, why not have a `static AtomicU64` that we increment every time to get the new ID?

Reply:
The POSIX shm namespace is global to the system, so that would collide with another window/application using softbuffer.

Reply:
Ah, I see. Still, using `rand` feels like overkill, especially given how many dependencies it comes with. Have you considered [`fastrand`](https://crates.io/crates/fastrand)?

Also, isn't there a way to create a "private" SHM segment through `IPC_PRIVATE`?

Reply:
`IPC_PRIVATE` is a System V shared memory flag, not a POSIX shared memory flag, so it's different. We can't use System V shared memory segments since they don't have file descriptors and can't be resized.

FreeBSD supports `SHM_ANON` in place of `name` for this, but it doesn't look like that's supported on any other platform and FreeBSD now offers a `memfd_create` function compatible with the Linux one. (Though I'm not sure when that was added, so this could be a concern for older FreeBSD versions).

Using a random name and a loop with `shm_open` and `shm_unlink` seems to be the standard approach. With variations. I see GTK's Wayland backend uses `sprintf (name, "/gdk-wayland-%x", g_random_int ());`, and the loop doesn't have any limit on the number of times it iterates.

If `rand` is a heavy dependency though, something else could be used. But I think some method of generating a random name is necessary.

Reply:
It doesn't look like `rand` shows up anywhere else in the `winit` dependency tree, so I'd be hesitant to add it. Like I said, I'd rather use a lighter dependency like `fastrand`.

Reply:
Switched to `fastrand`.